From 16296fcb4d66082d5c26a8f67f0ad17ed5ef9cf9 Mon Sep 17 00:00:00 2001
From: Lucas Yuki Imamura <lucasyuki@yahoo.com.br>
Date: Thu, 12 Jun 2025 19:18:48 +0000
Subject: [PATCH] odb: dbIStream clear containers

Signed-off-by: Lucas Yuki Imamura <lucasyuki@yahoo.com.br>
---
 src/odb/include/odb/dbStream.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/odb/include/odb/dbStream.h b/src/odb/include/odb/dbStream.h
index 46eabb056f9..e081b7209a8 100644
--- a/src/odb/include/odb/dbStream.h
+++ b/src/odb/include/odb/dbStream.h
@@ -393,6 +393,7 @@ class dbIStream
   {
     uint sz;
     *this >> sz;
+    m.clear();
     for (uint i = 0; i < sz; i++) {
       T1 key;
       T2 val;
@@ -407,6 +408,7 @@ class dbIStream
   {
     uint sz;
     *this >> sz;
+    m.clear();
     for (uint i = 0; i < sz; i++) {
       T1 key;
       T2 val;
@@ -421,6 +423,7 @@ class dbIStream
   {
     uint sz;
     *this >> sz;
+    m.clear();
     for (uint i = 0; i < sz; i++) {
       T1 key;
       T2 val;
@@ -436,6 +439,7 @@ class dbIStream
   {
     uint sz;
     *this >> sz;
+    m.clear();
     m.reserve(sz);
     for (uint i = 0; i < sz; i++) {
       T1 val;
